common/bitutil/compress.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2013 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package bitutil implements fast bitwise operations.
+package bitutil
+
+/*
+The compression algorithm implemented by CompressBytes and DecompressBytes is
+optimized for "sparse" input data which contains a lot of zero bytes. Decompression
+requires knowledge of the decompressed data length. Compression works as follows:
+
+if data only contains zeroes,
+  CompressBytes(data) == nil
+otherwise if len(data) <= 1,
+ CompressBytes(data) == data
+otherwise:
+ CompressBytes(data) == append(CompressBytes(nonZeroBits(data)), nonZeroBytes(data)...)
+where
+ nonZeroBits(data) is a bit vector with len(data) bits (MSB first):
```

Comment:
You could probably phrase this as "nonZeroBits(data) is a bit vector with len(data) bits, where the nth bit is set iff the nth byte of data is nonzero" without loss of clarity.